### ISSUE TITLE
Deprecated APIs with patch 11.0.0

### ISSUE THREAD

**Issue by lucasvienna** [Myrroddin/repbyzone#10]
While the TOC was bumped to the latest version, the deprecated APIs are still in use:

```lua
1x RepByZone/Core-Retail.lua:663: attempt to call upvalue 'GetFactionInfoByID' (a nil value)
[string "@RepByZone/Core-Retail.lua"]:663: in function <RepByZone/Core-Retail.lua:660>

Locals:
(*temporary) = nil
(*temporary) = 2510
(*temporary) = "attempt to call upvalue 'GetFactionInfoByID' (a nil value)"
watchedFactionID = 2510
factionName = nil
_ = nil
isWatched = nil
GetFactionInfoByID = nil
C_Reputation = <table> {}
```

I tried quickly replacing them with the newer [C_Reputation.GetFactionDataByID](https://warcraft.wiki.gg/wiki/API_C_Reputation.GetFactionDataByID), but it quickly turned into more than just search and replace, so I thought best to open an issue.

**Comment by Myrroddin** [comment]
You are correct, this is more than fixing a line or two with an updated API. Blizzard changed lots of things. I'm working on it.

**Comment by Myrroddin** [comment]
Fixed with version 1.34.17
